src/main/java/com/github/javabdd/BDD.java

```diff
@@ -586,6 +586,56 @@ public BDD relprod(BDD that, BDDVarSet var) {
      */
     public abstract BDD satOne(BDDVarSet var, boolean pol);
 
+    /**
+     * Computes the BDD that represents the set of successor states from {@code states}, i.e., the ones reachable by
+     * taking one forward step from states in {@code states}, where this BDD is the (partial) transition relation to
+     * consider. It is assumed that old-state and new-state BDD variables are interleaved in this transition relation.
+     *
+     * @param states The BDD representing the set of states.
+     * @param vars The BDD representing the set of variables that are relevant to consider for determining successor
+     * states. This variable set <i>must</i> include all variables that occur in the transition relation BDD. Moreover,
+     * for every new-state variable that is in {@code vars}, {@code relnext} considers the corresponding old-state
+     * variable to be relevant as well, regardless of whether it's included in {@code vars} or not.
```

Comment:
Does it not consider for old vars also the new vars? I think it does.

Reply:
If some old-state var `x` is in `vars` but the corresponding new-state variable `x'` is not, then `relnext`/`relprev`/... will not apply quantification and thus the value of `x` will not change.

If the new-state variable `x'` happens to be in `vars` but the old-state variable `x` is not, then still `relnext`/`relprev`/... may consider different values for `x`, hence it must be considered relevant. (If we want we could turn this into a precondition, and require that if `x'` is in `vars` then also `x` must be in `vars`. But the implementation now works fine also without this assumption. It just considers `x` relevant if `x'` is relevant.)

Reply:
OK, clear. I think we can leave it as is. It is properly documented in the JavaDoc.

Reply:
Maybe one more question: why does this not differ for the different directions (`relprev`/`relnext`)?

Reply:
The transition relation that's given to `relnext(..., relation, ...)`/`relprev(..., relation, ...)`/... stays the same regardless of the direction in which it is applied. In both cases, `relation` may e.g. quantify over some new-state variables, and then it's up to `relnext`/`relprev`/... to handle that correctly.